File: app/src/main/java/com/tokyonth/installer/activity/UninstallActivity.java
package com.tokyonth.installer.activity;

import android.os.Bundle;

import androidx.annotation.Nullable;

import com.tokyonth.installer.Constants;
import com.tokyonth.installer.R;
import com.tokyonth.installer.base.BaseActivity;
import com.tokyonth.installer.database.SQLiteUtil;
import com.tokyonth.installer.widget.CustomizeDialog;
import com.tokyonth.installer.utils.ShellUtils;
import com.tokyonth.installer.utils.GetAppInfoUtils;

import java.util.Objects;

public class UninstallActivity extends BaseActivity {

    @Override
    public int setActivityView() {
        return 0;
    }

    @Override
    public void initActivity(@Nullable Bundle savedInstanceState) {
        String pkgName = Objects.requireNonNull(getIntent().getDataString()).replace("package:", "");
        CustomizeDialog.getInstance(this)
                .setTitle(R.string.text_uninstall)
                .setMessage(getString(R.string.text_confirm_uninstall_app, GetAppInfoUtils.getApplicationNameByPackageName(this, pkgName)))
                .setPositiveButton(R.string.text_uninstall, (dialog, which) -> {
                    int result = ShellUtils.execWithRoot(Constants.UNINSTALL_COMMAND + pkgName);
                    String str = result == 0 ? getString(R.string.text_uninstall_complete) : getString(R.string.text_uninstall_failure);
                    showToast(str);
                    finish();
                })
                .setNeutralButton(getText(R.string.uninstall_dialog_disable), (dialog, which) -> {
                    int result = ShellUtils.execWithRoot(Constants.FREEZE_COMMAND + pkgName);
                    String appName = GetAppInfoUtils.getApplicationNameByPackageName(this, pkgName);
                    String str = result == 0 ? getString(R.string.freeze_app_name, appName) : getString(R.string.freeze_failure);
                    if (result == 0) {
                        SQLiteUtil.addData(this, pkgName);
                    }
                    showToast(str);
                    finish();
                })
                .setNegativeButton(R.string.dialog_btn_cancel, (dialog, which) -> finish())
                .create().show();

    }

}

File: app/src/main/java/com/tokyonth/installer/activity/UninstallActivity.kt
package com.tokyonth.installer.activity

import androidx.viewbinding.ViewBinding
import com.tokyonth.installer.R
import com.tokyonth.installer.install.UninstallCallback
import com.tokyonth.installer.install.UninstallTask
import com.tokyonth.installer.utils.ktx.string
import com.tokyonth.installer.utils.ktx.toast
import com.tokyonth.installer.utils.DialogUtils
import java.util.regex.Matcher
import java.util.regex.Pattern

class UninstallActivity : BaseActivity(), UninstallCallback {

    private lateinit var pkgName: String

    override fun setBinding(): ViewBinding? = null

    override fun initView() {

    }

    override fun initData() {
        intent.dataString?.let {
            val pattern: Pattern =
                Pattern.compile("([a-zA-Z_][a-zA-Z0-9_]*)+([.][a-zA-Z_][a-zA-Z0-9_]*)+")
            val matcher: Matcher = pattern.matcher(it)
            while (matcher.find()) {
                pkgName = matcher.group()
            }
        }

        if (pkgName.isNotEmpty()) {
            DialogUtils.unInstallDialog(this, pkgName) {
                if (it == DialogUtils.POSITIVE_BUTTON) {
                    UninstallTask(pkgName, this).start()
                } else {
                    finish()
                }
            }
        } else {
            toast(string(R.string.cannot_found_pkg))
        }
    }

    override fun onUninstallResult(resultCode: Int) {
        toast(
            if (resultCode == 0)
                string(R.string.text_uninstall_complete)
            else
                string(R.string.text_uninstall_failure)
        )
        finish()
    }

}
